internal/attestation/crafter/materials/materials.go

```diff
@@ -34,6 +36,32 @@ type crafterCommon struct {
 	input  *schemaapi.CraftingSchema_Material
 }
 
+type crafterUploader struct {
+	*crafterCommon
+	uploader casclient.Uploader
+}
+
+// Craft will calculate the digest of the artifact, simulate an upload and return the material definition
+func (i *crafterUploader) Craft(ctx context.Context, artifactPath string) (*api.Attestation_Material, error) {
```

Comment:
probably to make it clearer I'd not 1) expose this function (a.k.a downcase it) and 2) call it something else, there is no need to be called craft right, maybe upload?

Reply:
I would use craft, we upload there and we return attestation material there.